bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/resources/Telnet.java

```diff
@@ -180,20 +147,17 @@ public boolean read() throws IOException {
     public List<String> readLine(int timeOut) throws ReadException, IOException {
         readResultList.clear();
         long timeZero = System.currentTimeMillis();
-        long timeAfterTry = 0;
-        long timeTryiedToRead = 0;
         if (client.isConnected()) {
             readLineResult = "";
             int i = 1;
             while (i != -1) {
                 i = bufferedStream.available();
                 byte[] buffer = new byte[i];
+                // TODO this could block for ever?
```

Comment:
I do not yet see that? reader.isReady does not exist. And the current one has a timeout and will stop when no input is available. Your suggestion will always keep looping.

And you are right about the comment because we validate it to be available it will not block 👍 So I will remove the comment.